doc/architecture-decisions/0012-routecontroller-route-deletion-finalizer.md

```diff
@@ -0,0 +1,27 @@
+# 11. Routecontroller Route Deletion Finalizer
+
+Date: 2020-05-11
+
+## Status
+
+Accepted
+
+## Context
+Without a finalizer, we could rely on cascading deletion of child objects when Routes are deleted and `RequeueAfter` to rebuild the virtual services and services  from its routes every `ResyncInterval` seconds. `ResyncInterval` is currently set to 30 seconds.
+
+However, this doesn’t meet our SLO to handle changes within 10 seconds. Cascading deletes don’t work within 10 seconds, so meeting the SLO would require `ResyncInterval` to be less than 10 seconds, which seems unreasonable. 
```

Comment:
I spent some time looking for docs / things supporting this observed behavior. Wasn't able to find much. The most I found was this integration test in k8s, maybe it can be linked here?

https://github.com/kubernetes/kubernetes/blob/af67408c172630d59996207a2f3587ea88c96572/test/integration/garbagecollector/garbage_collector_test.go#L385-L392